R/recipe_run.R

```diff
@@ -387,6 +387,14 @@ recipe_data <- function(config, info, parameters, dest, instance) {
     info <- recipe_substitute(info, parameters)
   }
 
+  if (!is.null(info$environment_variables)) {
+    yml_path <- info$inputs[info$inputs$file_purpose == "orderly_yml",
+                            "filename"]
+    env_vars <- resolve_env(info$environment_variables,
```

Comment:
See comments in the test, but in general I wonder if resolve_env is the right thing to use here. For example it will silently let through things like

```
environment_variables:
  a: my_env_var
```

because it does not match the regular expression. Instead doing simply

```
lapply(info$environment_variables, Sys_getenv)
```

(within `Map` really so that the error handling can be done) will probably be what you want as *everything* in this list should be an environment variable.

The original design for `resolve_env` is to take a list where some elements _may_ be environment variables and resolve them - things were only counted if they matched the regex `^\\$[0-9A-Z_]+$` otherwise they were skipped over

Reply:
Sounds good - I've removed use of `resolve_env` now and we are using `Sys_getenv` directly.

Also dealt with the `$` being required/not mixup. Looks like `Sys.getenv` doesn't want vars with `$` prefixed anyway so I have removed this from the test code.